ntp-daemon/src/peer.rs

```diff
@@ -142,27 +142,30 @@ where
             system_poll,
         );
 
-        match accept {
-            Err(accept_error) => {
-                info!(?accept_error, "packet is not accepted");
-            }
-            Ok(_) => match result {
-                Ok(update) => {
-                    info!("packet accepted");
-                    let msg = MsgForSystem::Snapshot(self.index, self.reset_epoch, update);
-                    self.channels.msg_for_system_sender.send(msg).await.ok();
-                }
-                Err(IgnoreReason::KissDemobilize) => {
-                    info!("peer must demobilize");
-                    let msg = MsgForSystem::MustDemobilize(self.index);
-                    self.channels.msg_for_system_sender.send(msg).await.ok();
+        match result {
```

Comment:
bit of a crappy diff, but this inverts the matches: now we first check the result, do something with the kiss code if needed, then look at whether the message is accepted.